src/main/scala/org/broadinstitute/dsde/workbench/avram/api/AvramServlet.scala

```diff
@@ -1,69 +1,50 @@
 package org.broadinstitute.dsde.workbench.avram.api
 
-import java.util.logging.Logger
+import java.util.logging.{Level, Logger}
 
-import scala.concurrent.duration.Duration
+import io.circe.Encoder
+import io.circe.syntax._
 import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
-
 import org.broadinstitute.dsde.workbench.avram.Avram
 import org.broadinstitute.dsde.workbench.avram.dataaccess.SamUserInfoResponse
 import org.broadinstitute.dsde.workbench.avram.model.AvramException
+import org.broadinstitute.dsde.workbench.avram.util.AvramResult
+import org.broadinstitute.dsde.workbench.avram.util.AvramResult.unsafeRun
 
-import scala.concurrent.Await
-import scala.concurrent.Future
-import scala.util.{Failure, Success}
+import scala.concurrent.ExecutionContext
 
 
 trait AvramServlet {
 
-  implicit val executionContext = scala.concurrent.ExecutionContext.Implicits.global
+  implicit val executionContext: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
 
-  val log = Logger.getLogger(getClass.getName)
+  val log: Logger = Logger.getLogger(getClass.getName)
   private val samDao = Avram.samDao
   private val bearerPattern = """(?i)bearer (.*)""".r
 
-  // TODO: Change the structures of registerOnComplete and handleAuthenticationRequest. Not doing it
-  // TODO: for PR #18 because we're going to be switching to using IO with AvramResult and this will
-  // TODO: likely be changed around anyway.
+  def handleAuthenticatedRequest[T](request: HttpServletRequest, response: HttpServletResponse)
+                                   (f: SamUserInfoResponse => AvramResult[T])
+                                   (implicit encoder: Encoder[T]): Unit = {
+    def writeBody(body: T): Unit = {
+      response.setStatus(HttpServletResponse.SC_OK)
+      response.setContentType("application/json")
```

Comment:
We should probably set Content-Length in addition to Content-Type. Beyond that, I'm not sure what headers GAE standard automatically adds. I just pinged a Leo API and here are the response headers I got:
```
$ curl -I  -X GET --header 'Accept: application/json' --header 'Authorization: Bearer ya29.GmJHBh4RUa09lcANTKwGle6ha66vzNOtsDmkNfxE19jx2GF7A69O2QO0lW7z1i-oAkBrHhqhgN8gwObodH-YwDk2lRF27-epjj27zDnLoPBFm2wG5kWr9zv3XafZz1no6VLykg' 'https://notebooks.firecloud.org/api/clusters?includeDeleted=false'
HTTP/1.1 200 OK
Date: Wed, 31 Oct 2018 02:48:23 GMT
Server: akka-http/10.1.1
X-Frame-Options: SAMEORIGIN
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: authorization,content-type,accept,origin,x-app-id
Access-Control-Allow-Methods: GET,POST,PUT,PATCH,DELETE,OPTIONS,HEAD
Access-Control-Max-Age: 1728000
Content-Type: application/json
Content-Length: 22539
Set-Cookie: LeoToken=ya29.GmJHBh4RUa09lcANTKwGle6ha66vzNOtsDmkNfxE19jx2GF7A69O2QO0lW7z1i-oAkBrHhqhgN8gwObodH-YwDk2lRF27-epjj27zDnLoPBFm2wG5kWr9zv3XafZz1no6VLykg; Max-Age=3531; Path=/; Secure
Vary: Accept-Encoding
Via: 1.1 google
Alt-Svc: clear
```

I wonder if we should set CORS headers too -- not sure.

Reply:
Interestingly, it looks like `response.getWriter.write(...)` gives us Content-Length. Still need to look into CORS and other headers. Here's what we currently get for headers:
```
Content-Type: application/json
Server: Development/1.0
Date: Wed, 31 Oct 2018 13:42:37 GMT
Cache-Control: no-cache
Expires: Mon, 01 Jan 1990 00:00:00 GMT
Content-Length: 278
```
Looks like cache headers could use some attention.

Reply:
Ah ok cool. That looks good for now -- we can punt CORS till later.